src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1397,7 +1402,7 @@ def _validate_app_dates(app_start_date, app_end_date, cert_start_date, cert_end_
 # pylint: disable=inconsistent-return-statements
 def create_service_principal_for_rbac(
         # pylint:disable=too-many-statements,too-many-locals, too-many-branches
-        cmd, name=None, years=None, create_cert=False, cert=None, scopes=None, role='Contributor',
+        cmd, name=None, years=None, create_cert=False, cert=None, scopes=None, role=None,
```

Comment:
Need to detect whether `role` is explicitly set and show the warning if not. 

If `role='Contributor'` is used, it won't be possible to detect if `role` is provided by the user or as the default value. 

See https://stackoverflow.com/a/14749388

Reply:
Another option is to check `role.is_default`, and show the warning message accordingly. If `role` is assigned a default value, its type will be `knack.validators.DefaultStr` which has `is_default` field indicating whether it is the default value.